main/core/src/eval/Evaluator.scala

```diff
@@ -362,15 +382,302 @@ case class Evaluator(home: os.Path,
       }
     }
 
+    logEnd.foreach(_())
+
     multiLogger.close()
 
     (newResults, newEvaluated)
   }
 
   def resolveLogger(logPath: Option[os.Path], logger: Logger): Logger = logPath match{
     case None => logger
-    case Some(path) => MultiLogger(logger.colored, log, FileLogger(logger.colored, path, debugEnabled = true))
+    case Some(path) => MultiLogger(logger.colored, log, new FileLogger(logger.colored, path, debugEnabled = true))
   }
+
+  type Terminal = Either[Task[_], Labelled[Any]]
+  type TerminalGroup = (Terminal, Agg[Task[_]])
+
+  def evaluateParallel(
```

Comment:
At a high level, this method is probably too big. Could we break it up into 3-5 smaller methods? That may make the data and dependencies between the different portions of this algorithm more obvious, since we'd have to pass them from method to method in the parameter lists